build.py

```diff
@@ -23,4 +28,98 @@ def decrypt():
 @task(_checkdir)
 def encrypt():
     '''Encrypt secrets.dec.tf file.'''
-    os.system("ansible-vault encrypt --output secrets.enc secrets.dec.tf && rm -rf secrets.dec.tf")
\ No newline at end of file
+    os.system("ansible-vault encrypt --output secrets.enc secrets.dec.tf && rm -rf secrets.dec.tf")
+
+@task(_checkdir)
+def layer_dependency(summary='False',quiet='False'):
+    """Check layer dependency from the current layers"""
+    try:
+        remote_states = _layer_dependency(path.get_working_path())
+    except Exception as e:
+        print('Error getting deps:',e)
+
+    if summary.lower() == 'true':
+        deps = {'this': []}
+        for rs in remote_states:
+            deps['this'].append(remote_states[rs]['key'])
+        remote_states = deps
+
+    try:
+        if quiet == 'False': print("Note layer dependency is calculated using remote states.\nNevertheless, other sort of dependencies could exist without this kind of resources,\ne.g. if you relay on some resource created in a different layer and not referenced here.")
+        if len(remote_states) > 1:
+          if len(remote_states) > 2:
+            if quiet == 'False': print('Found these remote states:')
+          else:
+            if quiet == 'False': print('Found this remote state:')
+          print(json.dumps(remote_states, indent=1))
+        elif type(remote_states) == dict:
+          print(json.dumps(remote_states, indent=1))
+        else:
+          if quiet == 'False': print('No dependency detected.')
+    except Exception as e:
+        print('Error processing deps:',e)
+
+def _layer_dependency(directory):
+
+    config_files = ['config.tf']
+    config_file_contents = []
+
+    # #########################################
+    # get the config file contents
+    # #########################################
+    try:
+      for config_file in config_files:
+          with open(f"{directory}/{config_file}", 'r') as f:
+              config_file_contents.append(hcl2.load(f))
+    except Exception as e:
+        raise Exception(f"Can not open config files: {e}")
+
+    remote_states = {}
+
+    if len(config_file_contents)>0:
+      # #########################################
+      # remote states
+      # #########################################
+      # they have this shape
+      # data "terraform_remote_state" "eks-vpc" {
+      #   backend = "s3"
+      #   config = {
+      #     region  = var.region
+      #     profile = var.profile
+      #     bucket  = var.bucket
+      #     key     = "apps-prd/k8s-eks/network/terraform.tfstate"
+      #   }
+      # }
+      remote_state_resource_name = 'terraform_remote_state'
+      for data in config_file_contents:
+              if 'data' in data.keys():
```

Comment:
you could flatten all this structure by skipping the current value if it doesn't met the criteria

```suggestion
              if not 'data' in data.keys():
                  continue
```

the same for the rest of the `ifs` inside the loop

Reply:
great suggestion!